Source/Common/ArgumentLoader.cpp

```diff
@@ -5,47 +5,250 @@
 #include <FEXCore/fextl/string.h>
 #include <FEXCore/fextl/vector.h>
 
-#include "cpp-optparse/OptionParser.h"
 #include "git_version.h"
 
-#include <stdint.h>
-
 namespace FEX::ArgLoader {
-void FEX::ArgLoader::ArgLoader::Load() {
-  RemainingArgs.clear();
-  ProgramArguments.clear();
-  if (Type == LoadType::WITHOUT_FEXLOADER_PARSER) {
-    LoadWithoutArguments();
-    return;
+enum class ParserTypes : uint32_t {
+  String,
+  BoolTrue,
+  BoolFalse,
+  StringEnum,
+  StringArray,
+  Max,
+};
+
+struct OptionDetails {
+  std::string_view Arg;
+  FEXCore::Config::ConfigOption Option;
+  ParserTypes ParseType;
+};
+
+constexpr static OptionDetails ArgToOption[] = {
+#define ARG_TO_CONFIG
+#include <FEXCore/Config/ConfigOptions.inl>
+};
+
+const OptionDetails* FindOption(std::string_view Argument) {
+  for (auto& Arg : ArgToOption) {
+    if (Arg.Arg == Argument) {
+      return &Arg;
+    }
   }
 
-  optparse::OptionParser Parser {};
-  Parser.version("FEX-Emu (" GIT_DESCRIBE_STRING ") ");
-  optparse::OptionGroup CPUGroup(Parser, "CPU Core options");
-  optparse::OptionGroup EmulationGroup(Parser, "Emulation options");
-  optparse::OptionGroup DebugGroup(Parser, "Debug options");
-  optparse::OptionGroup HacksGroup(Parser, "Hacks options");
-  optparse::OptionGroup MiscGroup(Parser, "Miscellaneous options");
-  optparse::OptionGroup LoggingGroup(Parser, "Logging options");
+  return nullptr;
+}
+
+void PrintHelp() {
+  const char* Arguments[3] {};
+  Arguments[0] = "man";
+  Arguments[1] = "FEX";
+  Arguments[2] = nullptr;
+
+  execvp("man", (char* const*)Arguments);
```

Comment:
Why does it not print a built-in help for FEXLoader like it does for other tools in 4072?

Reply:
Decided to redirect to the man page because the options are complex enough with enough formatting issues that it would be a pain. It's becoming a common thing in Linux space that passing `-h` or `--help` in to Linux applications just load the man page anyway.